source/NuGet.Lucene.Web/Controllers/PackagesController.cs

```diff
@@ -90,12 +90,24 @@ public HttpResponseMessage DownloadPackage(string id, string version="")
                 return result;
             }
 
-            result = Request.CreateResponse(HttpStatusCode.OK);
+            if (Request.Headers.Range != null)
+            {
+                try
+                {
+                    HttpResponseMessage partialResponse = Request.CreateResponse(HttpStatusCode.PartialContent);
+                    partialResponse.Content = new ByteRangeStreamContent(package.GetStream(), Request.Headers.Range, new MediaTypeWithQualityHeaderValue("application/zip"));
+                    return partialResponse;
```

Comment:
If Chocalatey won't be sending `If-Range` we're probably getting into lily gilding.